pkg/operator/secretannotator/aws/reconciler.go

```diff
@@ -78,37 +115,61 @@ type ReconcileCloudCredSecret struct {
 	AWSClientBuilder func(accessKeyID, secretAccessKey []byte, c client.Client) (ccaws.Client, error)
 }
 
-// Reconcile will annotate the cloud cred secret to indicate the capabilities of the cred's capabilities:
+// Reconcile will typically annotate the cloud cred secret to indicate the capabilities of the cred's capabilities:
 // 1) 'mint' for indicating that the creds can be used to create new sub-creds
 // 2) 'passthrough' for indicating that the creds are capable enough for other components to reuse the creds as-is
 // 3) 'insufficient' for indicating that the creds are not usable for the cluster
+// In the event that the operator config resource has specified a mode to operate under (mint/passthrough)
+// then skip trying to determine the capabilities, and just annotate the secret.
 // +kubebuilder:rbac:groups=core,resources=secrets,verbs=get;list;watch;update
 func (r *ReconcileCloudCredSecret) Reconcile(request reconcile.Request) (reconcile.Result, error) {
 	start := time.Now()
 
-	r.Logger.Info("validating cloud cred secret")
-
-	operatorIsDisabled, err := utils.IsOperatorDisabled(r.Client, r.Logger)
+	mode, disabled, conflict, err := utils.GetOperatorConfiguration(r.Client, r.Logger)
 	if err != nil {
-		r.Logger.WithError(err).Error("error checking if operator is disabled")
+		r.Logger.WithError(err).Error("error checking operator configuration")
 		return reconcile.Result{}, err
-	} else if operatorIsDisabled {
-		r.Logger.Infof("operator disabled in %s ConfigMap", constants.CloudCredOperatorConfigMap)
+	} else if conflict {
+		r.Logger.Error("configuration conflict between legacy configmap and operator config")
+		return reconcile.Result{}, fmt.Errorf("configuration conflict")
+	} else if disabled {
+		r.Logger.Info("operator in disabled / manual mode")
 		return reconcile.Result{}, err
 	}
 
 	defer func() {
+		if err := r.syncOperatorStatus(); err != nil {
```

Comment:
Separate this into its own defer since the metrics recording is a distinct action not related to syncing the operator status.

Reply:
done